# Regressão Linear sem algorítmo

In [4]:
import pandas as pd
df  = pd.read_csv("C:/formacao_dataScience_DSA_DADOS/04_machineLearning/Cap04_Regressao_pt1/dados/pesos.csv")

In [5]:
df

Idade  Peso
0       0  3.44
1       1  4.39
2       2  4.49
3       3  4.78
4       4  5.23
5       5  6.00
6       6  6.18
7       7  7.03
8       8  7.18
9       9  7.50
10     10  8.59
11     11  8.99
12     12  9.54

Dado que estamos estimando o peso do bebê com base em sua idade, a regressão linear pode ser construída da seguinte maneira:

y = a + bx

Vamos resolver o problema a partir dos primeiros registros. Vamos supor que o conjunto de dados tenha apenas 2 pontos de dados. A modelagem ficaria assim:

Cálculo para o primeiro mês:
* y = a + b*x
* 3.44 = a + b*(0)
* a = 3.44

Cálculo para o segundo mês:
* y = a + b*x
* 4.39 = a + b*(1)
* 4.39 = 3.44 + b*(1)
* b = 4.39 - 3.44
* b = 0.95

As linhas anteriores já representam o treinamento do modelo e fomos capazes de prever os valores de a e b, nesse caso, a = 3.44 e b = 0.95. Isso é o que o modelo aprende durante o treinamento!

In [6]:
## RESUMO ESTATÍSTICO COM STATSMODELS

In [8]:
import statsmodels.formula.api as smf

In [10]:
# Criando modelo de Regressão
estimativa = smf.ols(formula= 'Peso ~ Idade', data = df)

In [15]:
# Lado esquerdo do "~" é variável dependente e a direita é independente
# POode ser até "exemplo ~ exemplo2 + exemplo3"

In [12]:
modelo = estimativa.fit()

In [13]:
print(modelo.summary())

                            OLS Regression Results                            
Dep. Variable:                   Peso   R-squared:                       0.986
Model:                            OLS   Adj. R-squared:                  0.985
Method:                 Least Squares   F-statistic:                     774.9
Date:                Fri, 24 May 2024   Prob (F-statistic):           1.50e-11
Time:                        09:47:12   Log-Likelihood:                 1.3432
No. Observations:                  13   AIC:                             1.314
Df Residuals:                      11   BIC:                             2.444
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      3.4737      0.124     27.937      0.0

c:\Users\alyss\anaconda3\lib\site-packages\scipy\stats\_stats_py.py:1769: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=13
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "
